include/nbl/video/utilities/CArithmeticOps.h

```diff
@@ -0,0 +1,286 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_VIDEO_C_ARITHMETIC_OPS_H_INCLUDED_
+#define _NBL_VIDEO_C_ARITHMETIC_OPS_H_INCLUDED_
+
+
+#include "nbl/video/IPhysicalDevice.h"
+#include "nbl/video/utilities/IDescriptorSetCache.h"
+
+#include "nbl/builtin/hlsl/scan/declarations.hlsl"
+
+
+namespace nbl::video
+{
+class CArithmeticOps : public core::IReferenceCounted
+{
+    public:
+		// Only 4 byte wide data types supported due to need to trade the via shared memory,
+		// different combinations of data type and operator have different identity elements.
+		// `EDT_INT` and `EO_MIN` will have `INT_MAX` as identity, while `EDT_UINT` would have `UINT_MAX`  
+		enum E_DATA_TYPE : uint8_t
+		{
+			EDT_UINT=0u,
+			EDT_INT,
+			EDT_FLOAT,
+			EDT_COUNT
+		};
+		enum E_OPERATOR : uint8_t
+		{
+			EO_AND = 0u,
+			EO_XOR,
+			EO_OR,
+			EO_ADD,
+			EO_MUL,
+			EO_MIN,
+			EO_MAX,
+			EO_COUNT
+		};
```

Comment:
instead of inventing new enums, I'd template `CArithmeticOps` using
```cpp
template<template<class> class Binop, typename T>
```
and determine what you're doing and summing that way